lib/pbench/agent/tool_meister.py

```diff
@@ -248,162 +348,133 @@ def wait(self):
+            f"{self.tool_inst_dir}/%Y%m%d.%H.%M",
+        ]
+
         return (0, "pcp tool (pmcd and pmlogger properly installed")
 
     def start(self):
-        assert self.tool_dir is not None, "Logic bomb!  no tool directory provided!"
-        if self.pmcd_process or self.pmlogger_process:
+        if self.pmcd_process is not None:
+            raise ToolException(
+                f"Tool({self.name}) has an unexpected pmcd process running"
+            )
+        if self.pmlogger_process is not None:
             raise ToolException(
-                f"Tool({self.name}) has an unexpected process still running"
+                f"Tool({self.name}) has an unexpected pmlogger process running"
             )
 
         self.logger.info(
-            "%s: start_tool -- %s -- %s",
+            "%s: start_tool -- '%s' && '%s'",
             self.name,
             " ".join(self.pmcd_args),
             " ".join(self.pmlogger_args),
         )
-        o_file = self.tool_dir / f"tm-{self.name}-start.out"
-        e_file = self.tool_dir / f"tm-{self.name}-start.err"
-        with o_file.open("w") as ofp, e_file.open("w") as efp:
-            try:
-                self.pmcd_process = subprocess.Popen(
-                    self.pmcd_args,
-                    cwd=self.tool_dir,
-                    stdin=subprocess.DEVNULL,
-                    stdout=ofp,
-                    stderr=efp,
-                )
-            except Exception as exc:
-                self.logger.error(
-                    "Pmcd run process failed: '%s', %r", exc, self.pmcd_args
-                )
-            try:
-                self.pmlogger_process = subprocess.Popen(
-                    self.pmlogger_args,
-                    cwd=self.tool_dir,
-                    stdin=subprocess.DEVNULL,
-                    stdout=ofp,
-                    stderr=efp,
-                )
-            except Exception as exc:
-                self.logger.error(
-                    "Pmlogger run process failed: '%s', %r", exc, self.pmlogger_args
-                )
+        self.pmcd_process = subprocess.Popen(
+            self.pmcd_args,
+            cwd=self.tool_inst_dir,
+            stdin=subprocess.DEVNULL,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        )
+        pmcd_reader = threading.Thread(
+            target=log_subprocess_output,
+            args=(self.pmcd_process.stdout, self.logger, self.__class__.__name__),
+        )
+        pmcd_reader.daemon = True
+        pmcd_reader.start()
+        self.pmlogger_process = subprocess.Popen(
+            self.pmlogger_args,
+            cwd=self.tool_inst_dir,
+            stdin=subprocess.DEVNULL,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        )
+        pmlogger_reader = threading.Thread(
+            target=log_subprocess_output,
+            args=(self.pmlogger_process.stdout, self.logger, self.__class__.__name__),
+        )
```

Comment:
This is going to send the PM logger output to the same log stream as the PMCD output...is that "OK"?

Reply:
Addressed in a new helper method.